src/pages/Swap/LimitOrderBox/components/OrderLimitPriceField/MarketPriceButton.tsx

```diff
@@ -0,0 +1,45 @@
+import { Currency } from '@swapr/sdk'
+
+import { memo } from 'react'
+import { useTranslation } from 'react-i18next'
+import styled from 'styled-components'
+
+import { ReactComponent as ProgressCircle } from '../../../../../assets/images/progress-circle.svg'
+import { MarketPrice } from './styles'
+
+export const MarketPriceButton = memo(
+  ({
+    buyTokenAmountCurrency: _,
+    sellTokenAmountCurrency: __,
+  }: {
+    buyTokenAmountCurrency: Currency
+    sellTokenAmountCurrency: Currency
+  }) => {
+    const { t } = useTranslation('swap')
+
+    const StyledProgressCircle = styled(ProgressCircle)`
+      width: 12px;
+      height: 12px;
+      margin-left: 4px;
+      transform: rotate(-90deg);
+
+      .move {
+        stroke-dasharray: 100;
+        stroke-dashoffset: 100;
+        animation: dash 15s 0s infinite linear forwards;
+
+        @keyframes dash {
+          to {
+            stroke-dashoffset: 0;
+          }
+        }
+      }
+    `
```

Comment:
Why is it in the component? Especially that it doesn't have props

Reply:
This is inside the component so we can rebuild the component and trigger the animation every time the component renders, otherwise the styled component doesn't render because is doesn't need.

Reply:
Also, this component receives props it doesn't use so it can re-render when props change. When this props change, the set to market price setInterval also is updated so the animation matches the interval.